mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -520,6 +533,10 @@ handle404 :: AWS a -> AWS (Maybe a)
 handle404 m = fmap Just m `catch` \ (e :: Error) ->
   if e ^? httpStatus == Just status404 then return Nothing else throwM e
 
+handle301 :: AWS a -> AWS (Maybe a)
+handle301 m = fmap Just m `catch` \ (e :: Error) ->
+  if e ^? httpStatus == Just status301 then return Nothing else throwM e
+
```

Comment:
(also vaguely recall that there was an amazonka bug with 301 handling that we did patch that hid the error - so hoping that this isn't related to a regression?)